tests/test_backends.py

```diff
@@ -0,0 +1,150 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import unittest
+import json
+import os
+import pytest
+import itertools
+
+from openqaoa.devices import DeviceQiskit, DeviceLocal, DeviceAWS
+from openqaoa.backends.qaoa_backend import get_qaoa_backend, DEVICE_NAME_TO_OBJECT_MAPPER, DEVICE_ACCESS_OBJECT_MAPPER
+from openqaoa.qaoa_parameters import Hamiltonian, create_qaoa_variational_params
+from openqaoa.qaoa_parameters.baseparams import QAOACircuitParams
+from openqaoa.utilities import X_mixer_hamiltonian
+from openqaoa.devices import create_device
+from openqaoa.basebackend import QAOABaseBackendShotBased
+
+from qiskit import IBMQ
+
+
+SUPPORTED_LOCAL_SIMULATORS = list(DEVICE_NAME_TO_OBJECT_MAPPER.keys())
+
+def get_params():
+    cost_hamil = Hamiltonian.classical_hamiltonian([[0, 1]], [1], constant=0)
+    mixer_hamil = X_mixer_hamiltonian(2)
+
+    circuit_params = QAOACircuitParams(cost_hamil, mixer_hamil, p=1)
+    variational_params_std = create_qaoa_variational_params(circuit_params, 'standard', 'ramp')
+
+    return circuit_params, variational_params_std
+
+class TestingBackendLocal(unittest.TestCase):
+    """
+    These tests check that the methods of the local backends are working properly.
+    """
+
+    def test_get_counts_and_expectation_n_shots(self):
+        """ 
+        Check that the .get_counts admit n_shots as an argument, and works properly for the backend of all local devices.
+        Also check that .expectation and .expecation_w_uncertainty methods admit n_shots as an argument for the QAOABaseBackendShotBased backends.
+        """
+
+        for device_name in SUPPORTED_LOCAL_SIMULATORS:
+
+            circuit_params, variational_params_std= get_params()
+
+            device = create_device(location='local', name=device_name)
+            backend = get_qaoa_backend(circuit_params=circuit_params, device=device, n_shots=1000)
+                        
+            assert sum(backend.get_counts(params=variational_params_std, n_shots=58).values())==58, "`n_shots` is not being respected for the local simulator `{}` when calling backend.get_counts(n_shots=58).".format(device_name)
+            if isinstance(backend, QAOABaseBackendShotBased): 
+                try: backend.expectation(params=variational_params_std, n_shots=58)
+                except Exception: raise Exception("backend.expectation does not admit `n_shots` as an argument for the local simulator `{}`.".format(device_name))
+                try: backend.expectation_w_uncertainty(params=variational_params_std, n_shots=58)
+                except Exception: raise Exception("backend.expectation_w_uncertainty does not admit `n_shots` as an argument for the local simulator `{}`.".format(device_name))
+
+class TestingBackendQPUs(unittest.TestCase):
```

Comment:
The tests are failing because of this test that I have implemented. Here I test that all backends allow n_shots as an attribute in .get_counts and .expectation methods.

I don't know if something like this should be done, and if so, how do put the credentials to test it, and how to make the github actions don't fail if there is no credentials.